src/steps/os/unix.rs

```diff
@@ -52,13 +53,26 @@ pub fn run_homebrew(cleanup: bool, run_type: RunType) -> Result<(), Error> {
 #[must_use]
 pub fn run_nix(run_type: RunType) -> Result<(), Error> {
     let nix = require("nix")?;
+    let nix_channel = require("nix-channel")?;
     let nix_env = require("nix-env")?;
 
     print_separator("Nix");
-    run_type.execute(&nix).arg("upgrade-nix").check_run()?;
+    if let Ok(Distribution::NixOS) = Distribution::detect() {
+        print_warning("Nix on NixOS must be upgraded via 'nixos-rebuild', skipping.");
```

Comment:
I rather that this will just be skipped without a warning. You can add a debug log if you like, however.